packit/upstream.py

```diff
@@ -305,41 +304,83 @@ def create_archive(
     ) -> str:
         return Archive(self, version).create(create_symlink=create_symlink)
 
-    def get_last_tag(self, before: str = None) -> Optional[str]:
+    def list_tags(self) -> List[str]:
         """
-        Get last git-tag from the repo.
-        :param before: get last tag before the given tag
+        List tags in the repository sorted by created date from the most recent.
         """
-
-        logger.debug(
-            f"We're about to `git-describe` the upstream repository "
-            f"{self.local_project.working_dir}."
-        )
-
         try:
-            version_tag_glob = self.package_config.upstream_tag_template.replace(
-                "{version}", "?*"
-            )
             cmd = [
                 "git",
-                "describe",
-                "--tags",
-                "--abbrev=0",
-                f"--match={version_tag_glob}",
+                "tag",
+                "-l",
+                "--sort=-creatordate",
             ]
-            if before:
-                cmd += [f"{before}^"]
-            last_tag = run_command(
+            tags = run_command(
                 cmd,
                 output=True,
                 cwd=self.local_project.working_dir,
-            ).stdout.strip()
+            ).stdout.split()
         except PackitCommandFailedError as ex:
             logger.debug(f"{ex!r}")
-            logger.info("Can't describe this repository, are there any git tags?")
-            # no tags in the git repo
+            logger.info("Can't list the tags in this repository.")
+            return []
+
+        return tags
+
+    def get_last_tag(
+        self, matching_only: bool = False, before: str = None
+    ) -> Optional[str]:
+        """
+        Get last git-tag from the repo.
+
+        Args:
+        matching_only: bool, whether to consider the `upstream_tag_include` and
+        `upstream_tag_exclude` regexes to filter out matching tags.
```

Comment:
Description of `before` is missing here, and I think the arguments should be indented more.